debezium-connect-rest-extension/src/main/java/io/debezium/kcrestextension/entities/PredicateDefinition.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.kcrestextension.entities;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.runtime.isolation.PluginDesc;
+import org.apache.kafka.connect.transforms.predicates.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * JSON model that describes a Single Message Transform (SMT) entry.
+ */
+public class PredicateDefinition extends PluginDefinition {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PredicateDefinition.class);
+
+    @JsonCreator
+    public PredicateDefinition(String className, ConfigDef config) {
+        super(className, config);
+    }
+
+    @JsonCreator
+    public static PredicateDefinition fromPluginDesc(PluginDesc<Predicate<?>> predicate) {
```

Comment:
`PluginDesc` is what we receive from Kafka Connect and I would personally say that changing it to avoid some depenendencies is an effort not really balanced with the improvements. We don't really have a dependency on connect-runtime we only consume a datastructure from there. That means we rely on data (what we do anyway) and not behavior. We would copy that datastructure 1:1 into a Debezium package. Is this really worth looking at the solid compatibility of Kafka packages? I see that for several reasons in some situations it is good to be decoupled from some external dependencies, but I do not know what is such a reason for Kafka/Kafka Connect deps (especially as a `provided` scope dependency)? I would also say that this architecture would increase the complexity of this PR from M to L/XL, wdyt? Is this really worth the effort now (in this PR)? @jpechane

Reply:
Hi @rk3rn3r and @jpechane, I do think there will be spots in the REST extension where we're inevitably going to have dependencies on the `connect-runtime` bits, for example, using the `Herder#plugins()` method to get the loaded transformations (unless we do our own classpath scanning of course).  

That said, as an alternative, what if we passed the `Class<?>` reference obtained from the `PluginDesc` as the argument?  That would remove the runtime dependency in this class's scope at the very least.

Obviously this all based on how we interact with `connect-runtime` in various spots, but if we can effectively centralize that into one or two classes at most, we guard ourselves from the module's wider view of code and if we face some compatibility concern in the future, the changes needed are far more isolated and targetable.  wdyt?

Reply:
I have removed the dependency from the connector modules. I have to keep it for this (the) general KC REST extension because I depend on the herder. I think this is okay because this code is purely made for KC. When we provide similar endpoints for DBZ Server in the future we won't have to re-use this. If we still want to re-use the same logic for DBZ Server we can rework this small part in the future. wdyt @Naros  @jpechane ?

Reply:
The rest-extension depends on debezium-core so I don't see any reason why we should not use DebeziumException for sake of uniformity.

Reply:
This modules name is `kcrestextension`. It is a KC specific implementation that adds some behavior which is missing from KC. Debezium Server (DS) might implement this separately and I don't see that we need to have this module compatible with DS.